### ISSUE TITLE
SecretsManager fail on node10.x

### ISSUE THREAD

**Issue by vuongngo** [aws/aws-sdk-js#2671]
Hi,

I upgraded lambda to nodejs10.x with aws-sdk (v2.457.0, v2.357.0). SecretsManager client  fail when trying to getSecretValue. This is an exception I received in cloudwatch.
```
ERROR	(node:7) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```
I reverted back to nodejs8.10 and it works fine (aws-sdk v2.457.0). Let me know if I miss anything when upgrading to nodejs10.x.

Thanks,

**Comment by trivikr** [comment]
The buffer constructor was deprecated in v10.0.0 ([details](https://nodejs.org/api/deprecations.html#deprecations_dep0005_buffer_constructor)), however we still use it at multiple places in SDK JS v2 

ToDo: Find a way to use alternative to Buffer constructor without breaking old supported versions of node

**Comment by trivikr** [comment]
A similar solution which was tried in the past can be viewed at https://github.com/aws/aws-sdk-js/blob/cb1604ca89a077ffdb86127884292d3b18c8b4df/lib/event-stream/to-buffer.js#L8-L11

**Comment by samparnell** [comment]
Any update on this? I upgraded to the Nodev10.x runtime yesterday and my logs are full of error messages from this. Seems to have reduced in the latest version of the SDK (2.463.0).

Main services used through the SDK: S3, CloudFront, DynamoDB, SNS

**Comment by trivikr** [comment]
@samparnell @vuongngo This issue will be fixed with release **v2.467.0** which will release tomorrow - 2019-05-31 at around 1200 hours PST

Do comment on this issue if bug still exists.

**Comment by samparnell** [comment]
@trivikr I pulled that down today and am testing it out now. I'm still seeing the error when I switch the lambdas over to v10. Going to test it out some more but on first pass it does _not_ appear to be resolved. 

Thanks

**Comment by AllanFly120** [comment]
@samparnell The sdk bundled in Lambda run time is not the newest. You might need to upload your [deployment package](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-create-deployment-pkg.html) with the newest sdk version. This is the sdk version that Lambda runtime is currently using: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

**Comment by trivikr** [comment]
Locking this thread as the issue is resolved in **v2.467.0**
Please open a [new issue](https://github.com/aws/aws-sdk-js/issues/new/choose) for related bugs and link to relevant comments in this thread.
